File: src/mandelbrot.frontend.guitest.runner/Runner.fs
namespace mandelbrot.frontend.guitest.runner

open iselenium
open NUnit.Framework
open OpenQA.Selenium.Edge
open System
open Microsoft.Extensions.Configuration

type Runner() =
    inherit SpaTest<EdgeDriver>()
    interface ITestServer with
        member val ServerProcesses = null with get, set
        member val driver = null with get, set

    member this.Config = this.GetConfig()

    [<SetUp>]
    member this.SetUp() =
        this.StartServer(this.Config,
            server=this.Config.["RendererServer"],
            root=this.Config.["RendererRoot"],
            port=Nullable<int>(this.Config.GetValue<int>("RendererPort")))
        this.StartServer()

    [<TearDown>]
    member this.TearDown() =
        this.StopServer()

    [<Test>]
    member this.RunTests() =
        this.Navigate("/test")

File: src/minimaltestrunner.core/Runner.cs
using iselenium;
using NUnit.Framework;
using OpenQA.Selenium.Edge;
using System.Collections.Generic;
using System.Diagnostics;

namespace minimaltestrunner
{
    [TestFixture]
    [Category("ITestServer")]
    public class Runner : SeleniumTest<EdgeDriver>, ITestServer
    {
        public List<Process> ServerProcesses { get; set; }

        [SetUp]
        public void SetUp()
        {
            this.StartServer();
        }

        [TearDown]
        public void TearDown()
        {
            this.StopServer();
        }

        [Test]
        public void RunTests()
        {
            this.Navigate("/");
            this.driver.Navigate().Refresh();
            this.ClickID("testButton");
            this.AssertTestsOK();
        }
    }
}